assets/src/edit-story/components/panels/captions/captions.js

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { v4 as uuidv4 } from 'uuid';
+import { useCallback } from 'react';
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { DropDown, Row, Button } from '../../form';
+import { ExpandedTextInput } from '../shared';
+import { SimplePanel } from '../panel';
+import { getCommonValue } from '../utils';
+import { useConfig } from '../../../app';
+import { useMediaPicker } from '../../mediaPicker';
+
+const Section = styled.div`
+  display: grid;
+  grid-template-areas: 'Options Remove';
+  grid-template-columns: auto 30px;
+`;
+const Options = styled.div`
+  grid-area: Options;
+`;
+const Remove = styled.div`
+  grid-area: Remove;
+  justify-self: center;
+  align-self: center;
+`;
+const RemoveButton = styled.button`
+  border: 0px;
+  background: none;
+  padding: 0;
+  color: ${({ theme }) => theme.colors.fg.white};
+`;
+
+function CaptionsPanel({ selectedElements, pushUpdate }) {
+  const tracks = getCommonValue(selectedElements, 'tracks', []);
+
+  const { languages } = useConfig();
+
+  const handleChangeLanguage = useCallback(
+    (inputValue, idToUpdate) => {
+      const trackIndex = tracks.findIndex(({ id }) => id === idToUpdate);
+      const languagesIndex = languages.findIndex(
+        ({ value }) => value === inputValue
+      );
+      const language = languages[languagesIndex];
+      const currentTrack = tracks[trackIndex];
+      const newTracks = [
+        ...tracks.slice(0, trackIndex),
+        { ...currentTrack, srclang: language.value, label: language.name },
+        ...tracks.slice(trackIndex + 1),
+      ];
+      pushUpdate({ tracks: newTracks }, true);
+    },
+    [tracks, pushUpdate, languages]
+  );
+
+  const handleChangeTrack = useCallback(
+    (attachment) => {
+      const newTracks = {
+        track: attachment?.url,
+        trackId: attachment?.id,
+        trackName: attachment?.filename,
+        id: uuidv4(),
+        kind: 'subtitles',
+        srclang: 'en',
+        label: __('English', 'web-stories'),
```

Comment:
This is the part where we'd need to use the user's current language I presume? Instead of English?